token/oauth/oauth2.go

```diff
@@ -122,21 +122,14 @@ func SaveReferrer(ctx context.Context, db application.DB, state uuid.UUID, refer
 // LoadReferrer loads referrer from DB
 func LoadReferrer(ctx context.Context, db application.DB, state string) (string, error) {
 	var referrer string
-	stateID, err := uuid.FromString(state)
-	if err != nil {
-		log.Error(ctx, map[string]interface{}{
-			"state": state,
-			"err":   err,
-		}, "unable to convert oauth state to uuid")
-		return "", err
-	}
-	err = application.Transactional(db, func(appl application.Application) error {
-		ref, err := appl.OauthStates().Load(ctx, stateID)
+
+	err := application.Transactional(db, func(appl application.Application) error {
+		ref, err := appl.OauthStates().Load(ctx, state)
 		if err != nil {
 			return err
 		}
 		referrer = ref.Referrer
-		err = appl.OauthStates().Delete(ctx, stateID)
+		err = appl.OauthStates().Delete(ctx, state)
```

Comment:
You don't have to change Delete(stateID) to Delete(state). You just loaded the reference and you have its ID (ref.ID). Then you won't need to load the reference in the Delete() again.

Reply:
@kishansagathiya can you take care of the above review comment?

Reply:
https://github.com/fabric8-services/fabric8-auth/pull/239/files#diff-12970846c32607659122a58e6cdb08c8R132
done here

Reply:
@sbose78 I think you were right about this.

Reply:
https://github.com/fabric8-services/fabric8-auth/pull/239/commits/56e8c2a968adcf3ed11bbbe754cf8c5b9dddc5f0